Content.Server/Station/Systems/StationSpawningSystem.cs

```diff
@@ -70,6 +75,22 @@ public override void Initialize()
             throw new ArgumentException("Tried to use a non-station entity as a station!", nameof(station));
 
         var ev = new PlayerSpawningEvent(job, profile, station);
+
+        if (station != null && profile != null)
+        {
+            switch (profile!.SpawnPriority)
+            {
+                case SpawnPriorityPreference.Arrivals:
+                    _arrivalsSystem.HandlePlayerSpawning(ev);
+                    _containerSpawnPointSystem.HandlePlayerSpawning(ev);
+                    break;
+                case SpawnPriorityPreference.Cryosleep:
+                    _containerSpawnPointSystem.HandlePlayerSpawning(ev);
+                    _arrivalsSystem.HandlePlayerSpawning(ev);
+                    break;
+            }
+        }
+
```

Comment:
1. This null suppression is unnecessary.
2. This should iterate every method of spawning and sort based on preference so it falls back to the other, then latejoin.